File: SozlukJava/app/src/main/java/com/example/sozlukjava/view/AnasayfaFragment.java
package com.example.sozlukjava.view;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.sozlukjava.R;
import com.example.sozlukjava.adapter.EntryAdapter;
import com.example.sozlukjava.databinding.FragmentAnasayfaBinding;
import com.example.sozlukjava.viewmodel.AnaSayfaViewModel;

// AnasayfaFragment.java
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.SearchView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.fragment.NavHostFragment;
import androidx.recyclerview.widget.LinearLayoutManager;

public class AnasayfaFragment extends Fragment {
    private FragmentAnasayfaBinding binding;
    private EntryAdapter recyclerAdapter;
    private AnaSayfaViewModel viewModel;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = FragmentAnasayfaBinding.inflate(inflater, container, false);
        binding.fabEntryEkle.setOnClickListener(v ->
                NavHostFragment.findNavController(this)
                        .navigate(R.id.action_anaSayfaFragment_to_entryEkleFragment)
        );
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        viewModel = new ViewModelProvider(this).get(AnaSayfaViewModel.class);
        binding.entryRecyclerView.setLayoutManager(new LinearLayoutManager(requireContext()));

        viewModel.loadMostCommentedEntriesToday();

        viewModel.getEntries().observe(getViewLifecycleOwner(), list -> {
            recyclerAdapter = new EntryAdapter(list,
                    entry -> {
                       AnasayfaFragmentDirections.ActionAnaSayfaFragmentToEntryDetayFragment action =
                               AnasayfaFragmentDirections.actionAnaSayfaFragmentToEntryDetayFragment(entry.getId());
                        NavHostFragment.findNavController(AnasayfaFragment.this).navigate(action);
                    },
                    entry -> { /* long click boş */ }
            );
            binding.entryRecyclerView.setAdapter(recyclerAdapter);
        });

        binding.searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override public boolean onQueryTextSubmit(String query) {
                viewModel.setSearchQuery(query != null ? query : "");
                return true;
            }
            @Override public boolean onQueryTextChange(String newText) {
                viewModel.setSearchQuery(newText != null ? newText : "");
                return true;
            }
        });

        binding.bottomNav.setOnItemSelectedListener(item -> {
            int id = item.getItemId();
            if (id == R.id.nav_bugun) {
                NavHostFragment.findNavController(this)
                        .navigate(R.id.action_anaSayfaFragment_to_bugunFragment);
                return true;
            } else if (id == R.id.nav_profil) {
                NavHostFragment.findNavController(this)
                        .navigate(R.id.action_anaSayfaFragment_to_profilFragment);
                return true;
            } else {
                return true; // Gündem zaten burası
            }
        });
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: SozlukKotlin/app/src/main/java/com/example/sozlukkotlin/view/AnasayfaFragment.kt
package com.example.sozlukkotlin.view

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.SearchView
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.lifecycleScope
import androidx.navigation.fragment.findNavController
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.sozlukkotlin.R
import com.example.sozlukkotlin.adapter.EntryAdapter
import com.example.sozlukkotlin.databinding.FragmentAnasayfaBinding
import com.example.sozlukkotlin.model.Entry
import com.example.sozlukkotlin.viewmodel.AnaSayfaViewModel
import kotlinx.coroutines.flow.launchIn
import kotlinx.coroutines.flow.onEach

class AnasayfaFragment : Fragment() {
    private var _binding: FragmentAnasayfaBinding? = null
    private val binding get() = _binding!!
    private lateinit var recyclerAdapter: EntryAdapter
    private lateinit var viewModel: AnaSayfaViewModel
    private var entryList: List<Entry> = listOf()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        _binding = FragmentAnasayfaBinding.inflate(inflater, container, false)
        binding.fabEntryEkle.setOnClickListener {
            findNavController().navigate(R.id.action_anaSayfaFragment_to_entryEkleFragment)
        }
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        viewModel = ViewModelProvider(this)[AnaSayfaViewModel::class.java]
        viewModel.loadMostCommentedEntriesToday()

        // RecyclerView ayarları
        binding.entryRecyclerView.layoutManager = LinearLayoutManager(requireContext())

        viewModel.entries.onEach { list ->
            recyclerAdapter = EntryAdapter(list , onClick = { entry ->
                val action = AnasayfaFragmentDirections.actionAnaSayfaFragmentToEntryDetayFragment(entry.id)
                findNavController().navigate(action)
            }, onLongClick = {} )
            binding.entryRecyclerView.adapter = recyclerAdapter
        }.launchIn(viewLifecycleOwner.lifecycleScope)

        binding.searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(query: String?): Boolean {
                viewModel.setSearchQuery(query.orEmpty())
                return true
            }

            override fun onQueryTextChange(newText: String?): Boolean {
                viewModel.setSearchQuery(newText.orEmpty())
                return true
            }
        })


        binding.bottomNav.setOnItemSelectedListener {
            when (it.itemId) {
                R.id.nav_bugun -> {
                    findNavController().navigate(R.id.action_anaSayfaFragment_to_bugunFragment)
                    true
                }
                R.id.nav_profil -> {
                    findNavController().navigate(R.id.action_anaSayfaFragment_to_profilFragment)
                    true
                }
                else -> true // Gündem zaten burası
            }
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
